src/lib/dnssd/ActiveResolveAttempts.h

```diff
@@ -159,17 +159,17 @@ class ActiveResolveAttempts
             case chip::Dnssd::DiscoveryFilterType::kNone:
                 return true;
             case chip::Dnssd::DiscoveryFilterType::kShortDiscriminator:
-                return browse.filter.code == static_cast<uint64_t>((data.nodeData.longDiscriminator >> 8) & 0x0F);
+                return browse.filter.code == static_cast<uint64_t>((data.Get<chip::Dnssd::CommissionNodeData>().longDiscriminator >> 8) & 0x0F);
```

Comment:
Consider taking `CommissionNodeData` reference once before the switch to reduce the overhead.